docs/ACCESSIBILITY.md

```diff
@@ -0,0 +1,139 @@
+
+The bot will then send another question with another set of answers.
+
+### Positive user experience
+
+We should disable buttons that are not part of the decision. Momentarily, the next set of answers will arrive. Since we must not change focus asynchronously outside of user gesture, the user is required to press <kbd>TAB</kbd> to get into the next set of decision buttons.
+
+When the user <kbd>TAB</kbd> out of the current button and into the next set of buttons, we should disable all previous decision buttons including the one the user chose. This will give a more consistent UX on how buttons are disabled.
+
+Since disabling buttons will also hide them from screen reader, we should add a screen reader-only text to tell the user which answer they chose.
+
+### Exceptions
+
+-  If no tabbable UI after the current disabled button, the next <kbd>TAB</kbd> should goes into the send box.
+
+### Implementation
+
+When an UI is being disabled:
+
+-  All UI will be manually styled, based on `:disabled, [aria-disabled="true"]` query
+   -  User agent stylesheet do not take account into `aria-disabled` attribute
+-  Set `aria-disabled` attribute to `true`
+   -  If it is a `<button>`, set `onClick` to a handler that call `event.preventDefault()`
+   -  If it is a `<input type="text">` or `<textarea>`, set `readOnly` to `true`
+-  If it is currently focused, wait until `onBlur` event and set `disabled` attribute
+   -  Otherwise, set `disabled` attribute immediately
+
+> List of elements support `disabled` attribute can be found in [this article](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled).
+
+### Additional context
+
+By default, HTML is static. Thus, the default `disabled` implementation works on a static web page.
+
+On a dynamic web page, when `disabled` is being applied to a focusing element (`document.activeElement`), the focus change varies between browsers:
+
+-  Chrome will keep the focus on current element invisible to JavaScript or CSS
+   -  On <kbd>TAB</kbd>, it jump to next tabbable sibling or descendants of them (depth-first search)
+-  Firefox will send the focus to the parent, invisibly
+   -  On <kbd>TAB</kbd>, it jump to the first tabbable sibling or descendants of them (depth-first search)
+-  Edge UWP and IE11 will send the focus to `<body>`, invisibly
+   -  On <kbd>TAB</kbd>, it jump to the first tabbable on the web page
+
+## UX: New messages button
+
+> This is related to [#3136](https://github.com/microsoft/BotFramework-WebChat/issues/3136).
+
+### User story
+
+While the user scrolled up to view past conversation, the bot send a message with few decision buttons to the user. To make the user aware of the arriving message, a "New messages" button is shown on the screen.
+
+The user should be able to <kbd>TAB</kbd> into the "New messages" button. Clicking on it will scroll the view to the first decision button and put the focus on it.
```

Comment:
```suggestion
The user should be able to move the focus to the "New messages" button by pressing <kbd>TAB</kbd>. Clicking on it will scroll the view to the first decision button and put the focus on it.
```